### ISSUE TITLE
Bug: Dark Mode Toggle is missing visible focus via keyboard only navigation.

### ISSUE THREAD

**Issue by MillyCodes** [mdn/yari#5480]
# Summary
Dark mode toggle in the new header is missing visible focus via keyboard only navigation.

## Where did you spot the bug?
[https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog)

## What is the problem?
At first thought, dark mode switch in header was not accessible via keyboard only, not reachable or toggle-able. Upon further review, it is reachable with keyboard but as a user, **there is no indication of focus** thus user may not understand if/how to properly interact with it. **Note:** when using Mac Voiceover a box does appear over the toggle that conveys some functionality and the toggle is announced properly, see screenshot below.

## What did you expect to happen?
The visible focus is observed on all the other interactive navigation elements but the dark mode toggle is missing this. ~It looks like this element has been completely removed from the focus order~ BUT I tried activating it, and it did work, so just missing some sort of focus/outline.

## Steps to reproduce the issue

1. Open https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog on chrome or firefox.
2. Using the keyboard ONLY, tab to the dark mode switch.
3. After navigating pass left side menu, focus lands on toggle but there is no visible indication.
4. You can't tell visually but focus is on toggle, hit SpaceBar the toggle functions properly.

### Device
- [x] Desktop

### Browser

- [x] Chrome: Version 98.0.4758.109 (Official Build) (x86_64)
- [x] Firefox: 97.0.1 (64-bit)

### Operating system
- [x] Mac OS: Catalina 10.15.7

## Screenshots
Visible indicator is missing but here you go.. 😅
![Screen Shot 2022-03-04 at 3 39 38 PM](https://user-images.githubusercontent.com/20953033/156838343-bf78e81f-0ac8-45b7-8efd-f28b85cd0301.png) ![Screen Shot 2022-03-04 at 3 39 52 PM](https://user-images.githubusercontent.com/20953033/156838348-13834b1c-e471-48ea-9000-8b0e717b773e.png)
Screenshot of toggle when using Voiceover.
![Screen Shot 2022-03-04 at 3 44 18 PM](https://user-images.githubusercontent.com/20953033/156839455-f440749a-668f-45ba-9116-99b901d09413.png)

**Comment by schalkneethling** [comment]
Fixed by https://github.com/mdn/yari/pull/5506
